apps/cnspec/cmd/serve.go

```diff
@@ -0,0 +1,185 @@
+package cmd
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.mondoo.com/cnquery"
+	"go.mondoo.com/cnquery/cli/config"
+	"go.mondoo.com/cnquery/cli/execruntime"
+	"go.mondoo.com/cnquery/cli/inventoryloader"
+	"go.mondoo.com/cnquery/cli/prof"
+	"go.mondoo.com/cnquery/cli/sysinfo"
+	"go.mondoo.com/cnquery/motor/asset"
+	v1 "go.mondoo.com/cnquery/motor/inventory/v1"
+	"go.mondoo.com/cnquery/motor/providers"
+	"go.mondoo.com/cnquery/resources"
+	"go.mondoo.com/cnquery/upstream"
+	"go.mondoo.com/cnspec"
+	"go.mondoo.com/cnspec/apps/cnspec/cmd/backgroundjob"
+	cnspec_config "go.mondoo.com/cnspec/apps/cnspec/cmd/config"
+	"go.mondoo.com/cnspec/policy/scan"
+	"go.mondoo.com/ranger-rpc"
+)
+
+// we send a 78 exit code to prevent systemd from restart
+// NOTE: if we change the code here we also need to adapt the systemd service
+const ConfigurationErrorCode = 78
+
+func init() {
+	rootCmd.AddCommand(serveCmd)
+	// background scan flags
+	serveCmd.Flags().Int("timer", 60, "scan interval in minutes")
+	serveCmd.Flags().MarkHidden("timer")
+}
+
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Start cnspec in background mode",
+
+	PreRun: func(cmd *cobra.Command, args []string) {
+		viper.BindPFlag("timer", cmd.Flags().Lookup("timer"))
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		prof.InitProfiler()
+
+		// prevent colors on windows
+		viper.Set("color", "none")
+
+		// check if an inventory file exists
+		inventoryFilePath, ok := config.InventoryPath(viper.ConfigFileUsed())
+		if ok {
+			viper.Set("inventory", inventoryFilePath)
+		}
+
+		// determine the scan config from pipe or args
+		conf, err := getServeConfig()
+		if err != nil {
+			log.Fatal().Err(err).Msg("failed to prepare config")
+		}
+
+		ctx := cnquery.SetFeatures(context.Background(), cnquery.DefaultFeatures)
+
+		if conf != nil && conf.UpstreamConfig != nil {
+			hc := backgroundjob.NewHealthPinger(ctx, conf.UpstreamConfig.ApiEndpoint, 5*time.Minute)
```

Comment:
This is the health check so that the user can see that the client is still running. It just pings the server and says: Hey I am here. We identified this issue in the past where users quickly wanted to identify if all their services are still running.